src/coreclr/pal/src/thread/context.cpp

```diff
@@ -1569,6 +1487,65 @@ CONTEXT_SetThreadContextOnPort(
         }
     }
 
+    if (lpContext->ContextFlags & (CONTEXT_CONTROL|CONTEXT_INTEGER) & CONTEXT_AREA_MASK)
```

Comment:
It may be useful to add a note that the integer/control registers have to be set last.